client/settings/settings-sub-page.ts

```diff
@@ -0,0 +1,43 @@
+import { TFunction } from 'i18next'
+import { assertUnreachable } from '../../common/assert-unreachable'
+
+export enum AppSettingsSubPage {
+  Sound = 'AppSound',
+  System = 'AppSystem',
+}
+
+export enum GameSettingsSubPage {
+  StarCraft = 'StarCraft',
+  Input = 'GameInput',
+  Sound = 'GameSound',
+  Video = 'GameVideo',
+  Gameplay = 'Gameplay',
+}
+
+export type SettingsSubPage = AppSettingsSubPage | GameSettingsSubPage
+
+export const ALL_SETTINGS_SUB_PAGES: ReadonlyArray<SettingsSubPage> = [
+  ...Object.values(AppSettingsSubPage),
+  ...Object.values(GameSettingsSubPage),
+]
+
+export function settingsSubPageToLabel(subPage: SettingsSubPage, t: TFunction): string {
```

Comment:
Don't we think this is an anti-pattern? This should most likely just be a component somewhere

Reply:
Hmm, I think this is fine to have as a function; we have plenty of other places where we use something like this (e.g. `matchmakingTypeToLabel`).

The problem of trying to make this into a component is that linter complains if you return only a `string` from a component, so you would have to wrap it in some HTML, I think. And then you would have to style that HTML differently based on where it's used. I guess it's not that big of a deal, but I also don't particularly see what's anti-patterny about this approach either.

Reply:
Okay let me rephrase: this is an anti-pattern and it should be a component somewhere. The only time we need functions like this is if we need a label in a non-React context, and I really struggle to think of anything where that would be the case. Components such as this should retrieve the `t` themselves.

It is not a problem to need to style things differently, just take a `className` prop and use styled-components as normal.

Reply:
For context, all of the functions like this exist because I wrote them before we knew what localization strategy we were going to use, and I wanted consistent strings for various things that we could easily know to localize once we got there. We're past that point now, so we can actually write things that are appropriate for what we're using (which is: write a React component and use the relevant hooks).